### DISCUSSION TITLE
[Bug] "Add file" > "Upload files" fails with HTTP error 400 on commit if upload is not finished yet

### DISCUSSION THREAD

**Discussion by Marcono1234** [community/community#51683] [Repositories]
### Select Topic Area

Bug

### Description

Trying to add files via "Add file" > "Upload files" fails with a HTTP 400 error on commit if the upload is not finished yet.

Browser: Firefox 111.0.1 (64-bit), 153.0.3 (64-Bit)
Last tested: 2026-08-12

### Reproduction steps

1. On a the top right of the main page of a repository, click "Add file", then choose "Upload files"
2. Open the Developer Tools of your browser and there open the "Network" tab (needed for step 4)
  (Optional: In the Developer Tools [throttle the network speed](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/throttling/index.html); remember to revert again.)
3. Drag any file which is a few MBs large in the upload area of the GitHub page
:x: Bug 1: Even though the upload is still in progress the "Commit" button is not disabled
(this differs from for example uploads while creating a GitHub Issue where the "Submit new issue" button is disabled during upload)
4. While the upload is still in progress click "Commit"
:x: Bug 2: A completely blank page opens; the "Network" tab of the browser Developer Tools shows that a request to `https://github.com/<user>/<repo>/upload` failed with HTTP 400 Bad Request

**Comment by ilovecats4606** [comment]
3 years later, this issue is still present.

**Reply by Hedwig207** [reply]
bro每次都急不可耐的点提交，结果回回开github...

**Comment by tanvishinde017** [comment]
Thanks for the detailed reproduction steps.

I was able to follow the reported workflow conceptually, and the behavior described does appear inconsistent with other GitHub upload experiences. If the file upload has not completed, the **Commit changes** button should ideally remain disabled or provide clear feedback that the upload is still in progress.

Based on your report, there seem to be two separate issues:

1. **UI/UX issue:** The Commit button remains available while the upload is still being processed.
2. **Error handling issue:** Submitting before upload completion results in a blank page and an HTTP 400 response instead of a user-friendly validation message.

Expected behavior would be:

* The Commit button is disabled until all uploads finish successfully.
* If a commit is attempted prematurely, GitHub should display a clear error message and keep the user on the upload page rather than navigating to a blank page.

The inclusion of browser version, network observations, and precise reproduction steps is very helpful for investigation. If possible, it may also be useful to include:

* Repository visibility (public/private)
* Approximate file size
* Whether the issue reproduces in another browser
* Screenshots of the upload progress and resulting blank page

Hopefully the GitHub team can reproduce and address both the premature commit action and the resulting error-handling behavior.

**Comment by ShubhamTiwari007xx** [comment]
Disable the Commit changes button whenever file uploads are in progress and re-enable it only after all uploads finish successfully. This prevents invalid requests and improves user experience.
